sw/device/lib/testing/test_framework/ottf_sival.h

```diff
@@ -0,0 +1,45 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_TEST_FRAMEWORK_OTTF_SIVAL_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_TEST_FRAMEWORK_OTTF_SIVAL_H_
+
+#include <stdbool.h>
+#include <stdnoreturn.h>
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/testing/json/command.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf_commands.h"
+
+/**
+ * Wait until a condition is true.
+ *
+ * @param cond: an expression that can be repeated tested for.
+ * @param timeout_usec: timeout in microseconds.
+ *
+ * In DV, this function will simply spin since the DV environment can do
+ * backdoor writes to make the condition true. On real device, this will wait
+ * for host to send ujson message. The condition is check after each command.
+ */
+#define OTTF_SIVAL_WAIT_UNTIL(cond, timeout_usec)                          \
```

Comment:
nit: I would name this macro `OTTF_WAIT_FOR` since it will work for all test targets including `kDeviceSimDV`.